src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -250,7 +249,20 @@ public SAMSequenceDictionary getReferenceDictionary() {
      * @return best available sequence dictionary given our inputs
      */
     public SAMSequenceDictionary getBestAvailableSequenceDictionary() {
-        return reference != null ? reference.getSequenceDictionary() : (reads != null ? reads.getSequenceDictionary() : null);
+        if (hasReference()){
+            reference.getSequenceDictionary();
+        }
+        if (hasReads()){
+            return reads.getSequenceDictionary();
+        }
+        if (hasFeatures()){
+            final List<SAMSequenceDictionary> dictionaries = features.getSequenceDictionaries();
+            //If there is just one, it clearly is the best. Otherwise, noone is best.
+            if (dictionaries.size() == 1){
+                return dictionaries.get(0);
+            }
```

Comment:
Hmm, interesting choice -- is there nothing we can do when there are multiple variant headers and no reference/reads? Eg., could we use something like `VCFUtils.smartMergeHeaders()`, or do you think the convenience of auto-merging would be outweighed by the inconvenience of getting an exception when the headers are incompatible and can't be merged?